src/story/comment.py

```diff
@@ -0,0 +1,152 @@
+import json
+import sys
+
+sys.path.append("..")
+
+from app import sql_db
+from datetime import datetime
+from flask import request
+from flask_restful import reqparse, abort, Api, Resource
+from src.auth.login_manager import access_token_required, id_token_required
+from src.util import generate_response
+from src.data_models.comment_model import CommentModel
+
+parser = reqparse.RequestParser()
+parser.add_argument("story_id")
+parser.add_argument("user_id")
+parser.add_argument("comment_id")
+
+
+def get_comments(article_id, user_id, comment_id):
+    """Get comments filtered by article_id, user_id, comment_id if available.
+
+    Send request to SQL database to get comments based on article_id,
+    user_id, comment_id. If an ID is not available, then skip this when
+    filtering.
+
+    Args:
+        article_id: int, unique article id
+        user_id: int, unique user id
+        comment_id: int, unique comment id
+    Return:
+        A list of comment dictionary.
+    """
+    args = {}
+    if article_id:
+        args["article_id"] = article_id
+    if user_id:
+        args["user_id"] = user_id
+    if comment_id:
+        args["comment_id"] = comment_id
+    comments = CommentModel.query.filter_by(**args).all()
+    return [comment.to_dict() for comment in comments]
+
+
+class Comment(Resource):
+    @access_token_required
+    def get(self):
+        """Get comments from SQL database.
+
+        Args from request:
+            story_id: int, unique article id
+            user_id: int, unique user id
+            comment_id: int, unique comment id
+        Return:
+            A list of comment dictionary.
+        """
+        request.get_json()
+        args = parser.parse_args()
+        if args["story_id"] is None:
+            return generate_response("Error: Story ID not found. ", 400)
+        return generate_response(
+            get_comments(args["story_id"], args["user_id"], args["comment_id"]), 200
+        )
+
+    @id_token_required
+    def post(okta_id, user_id, self):
+        """Create a new comment post.
+
+        Args from request:
+            story_id: int, unique article id
+            user_id: int, unique user id
+            comment_content: str, the content of comment(in request body)
+        Return:
+            A response specifying whether the comment creation is successed.
+        """
+        args = parser.parse_args()
+        if args["story_id"] is None or args["user_id"] is None:
+            return generate_response("Error: Invalid parameters.", 400)
+        if args["user_id"] != str(user_id):
+            return generate_response("Error: Not Authorized.", 403)
+        request_data = json.loads(request.get_data())
```

Comment:
we get error `json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)` if there is no request body, should we add a error response handler for that case?

Reply:
Fixed. Add an exception handler for the error request body. Thanks for the suggestion!